gen/service.go

```diff
@@ -280,3 +286,37 @@ func functionUnwrapResponse(g Generator, f *compile.FunctionSpec) (string, error
 			}
 		`, f)
 }
+
+func functionArgsEnveloper(g Generator, f *compile.FunctionSpec) error {
+	return g.DeclareFromTemplate(
+		`
+		<$wire := import "github.com/thriftrw/thriftrw-go/wire">
+		<$argType := printf "%v%v" (goCase .Name) "Args">
+		<$v := newVar "v">
+
+		func (<$v> *<$argType>) MethodName() string {
+			return "<.MethodName>"
+		}
+
+		func (<$v> *<$argType>) Type() <$wire>.EnvelopeType {
```

Comment:
This makes it so `type` and `methodname` (and other-cased variants) can't be used as parameter names. `type` is already reserved by the namespacer, but `methodname` is not. You should probably add a TODO somewhere about handling that conflict (probably in the same place with the TODO for `towire/fromwire` conflicts).